jetty-http/src/main/java/org/eclipse/jetty/http/pathmap/RegexPathSpec.java

```diff
@@ -21,6 +21,108 @@
 import org.eclipse.jetty.util.log.Log;
 import org.eclipse.jetty.util.log.Logger;
 
+/**
+ * <p>
+ * RegexPathSpec is a PathSpec implementation for a {@link PathMappings} instance.
+ * </p>
+ *
+ * <p>
+ * Supports the standard Java regex found in {@link java.util.regex.Pattern}.
+ * </p>
+ *
+ * <p>
+ * Supports {@link PathSpecGroup} for {@link PathSpecGroup#EXACT}, {@link PathSpecGroup#PREFIX_GLOB}, {@link PathSpecGroup#MIDDLE_GLOB}, and {@link PathSpecGroup#SUFFIX_GLOB}.
+ * This is done by evaluating the signature or the provided regex pattern for what is a literal vs a glob (of any kind).
+ *     <ul>
+ *         <li>Only literals, it's a {@link PathSpecGroup#EXACT}.</li>
+ *         <li>Starts with literals, ends with globs, it's a {@link PathSpecGroup#PREFIX_GLOB}</li>
+ *         <li>Starts with glob, has at least 1 literal, then any thing else, it's a {@link PathSpecGroup#SUFFIX_GLOB}</li>
+ *         <li>All other signatures are a {@link PathSpecGroup#MIDDLE_GLOB}</li>
+ *     </ul>
+ *     The use of regex capture groups, regex character classes, regex quantifiers, and regex special contructs
+ *     will be identified as a glob (for signature determination), all other characters are identified
+ *     as literal.  The regex {@code ^} beginning of line, and regex {@code $} end of line are ignored.
+ * </p>
+ *
+ * <p>
+ *    <b>Support for {@link MatchedPath} and PathMatch vs PathInfo</b>
+ * </p>
+ *
+ * <p>
+ *     There's a few steps in evaluating the matched input path for determining where the split
+ *     in the input path should occur for {@link MatchedPath#getPathMatch()} and {@link MatchedPath#getPathInfo()}.
+ *     <ol>
+ *         <li>If there are no regex capturing groups,
+ *           the entire path is returned in {@link MatchedPath#getPathMatch()},
+ *           and a null returned for {@link MatchedPath#getPathInfo()}</li>
+ *         <li>If the named regex capturing group {@code name} is present, that group
+ *           is returned in {@link MatchedPath#getPathMatch()}</li>
+ *         <li>If the named regex capturing group {@code info} is present, that group
+ *           is returned in {@link MatchedPath#getPathInfo()}</li>
+ *         <li>If the regex is of group type {@link PathSpecGroup#PREFIX_GLOB},
+ *           the beginning of the regex is a literal, so it is split at the start of
+ *           {@code java.util.regex.Matcher.group(1)}),
+ *           taking care to handle trailing slash properly so that {@link MatchedPath#getPathMatch()}
+ *           does not end in it, and {@link MatchedPath#getPathInfo()} starts with it.</li>
```

Comment:
I think this correctly describes the current behavior (although I'm still concerned about having more than 1 group).  However, I think we could perhaps consider changing behavior to:

```suggestion
 *         <li>If there is only one capturing group, it is returned as the  {@link MatchedPath#getPathInfo()} and any characters prior to the start of the capture group are returned as the {@link MatchedPath#getPathMatch()}.</li>
```

Or perhaps to better align with the servlet spec

```suggestion
 *         <li>If there is only one anonymous capturing group, the path is split at the start of the capture and the first part of the split is returned as  the {@link MatchedPath#getPathMatch()} and the second part is returned as the  {@link MatchedPath#getPathInfo()}. The first part may be the empty string and the second part may be null</li>
```

Note that this doesn't depend on the type of the match.  I.e we could split an exact match this way.  Why not?

We could define what to do with 2 capture groups, but I don't really see the need.